t/lifecycle.t

```diff
@@ -112,6 +112,48 @@ subtest 'Everything but Test::NoWarnings' => sub {
 
             is count_cancellation_messages( $db->test_results( $testid4 ) ), 1, 'one cancellation message present after crash';
         };
+
+        subtest 'Do not reuse batch tests' => sub {
+            my %user = (
+                username => "user",
+                api_key  => "key"
+            );
+            my @domains = ( 'zone1.rpcapi.example', 'zone5.rpcapi.example' );
+            my $params = {
+                %user,
+                domains => \@domains,
+                test_params => {
+                    priority => 5,
+                    queue   => 0
+                }
+            };
+            $db->add_api_user( $user{username}, $user{api_key} );
+            my $batch_id = $db->add_batch_job( $params );
+
+            my @batch_test_ids = $db->dbh->selectall_array(
+                q[
+                    SELECT hash_id
+                    FROM test_results
+                    WHERE batch_id = ?
+                ],
+                undef,
+                $batch_id
+            );
+            @batch_test_ids = map { $$_[0] } @batch_test_ids;
+
+            is( @batch_test_ids, 2, 'two test created in batch' );
```

Comment:
This seems to be more of a sanity check than directly connected to the purpose of the subtest. Maybe replace `is` with `BAIL_OUT`?

Reply:
Done.